src/component/pages/openam.js

```diff
@@ -0,0 +1,100 @@
+
+export default function OpenAM({setAccessToken = f => f, error, setError = f => f}) {
+   const location = useLocation();
+   const history = useHistory();
+   const [isTokenLoaded, setTokenLoaded] = useState(false);
+
+   async function getOAuthTokens(refreshToken) {
+      const credentials  = REACT_APP_CLIENT_ID + ':' + REACT_APP_CLIENT_SECRET;
+      const authorizationHeader = "Basic " + new Buffer(credentials).toString("base64");
+      const client = axios.create({ headers: { "Authorization": authorizationHeader, "Content-Type": "application/x-www-form-urlencoded" } });
+
+      const data = [
+         `grant_type=refresh_token`,
+         `refresh_token=${refreshToken}`,
+         `realm=external`
+      ].join("&");
+
+      const response = await client.post(REACT_APP_OAUTH_PROVIDER_URL + "/access_token", data);
+      const { access_token, refresh_token } = response.data;
+
+      Cookie.set("igOpenAMRefreshToken", refresh_token);
+      return { access_token, refresh_token };
+   }
+
+   function redirectToOpenAM() {
+      const state = REACT_APP_CLIENT_STATE || 'af0ifjsldkj';
+      const qParams = [
+         `response_type=code`,
+         `client_id=${REACT_APP_CLIENT_ID}`,
+         `redirect_uri=http://${window.location.hostname}:${window.location.port}/oauth`,
+         `scope=full_readonly`,
+         `realm=external`,
+         `state=${state}`
+      ].join("&");
+
+      window.location.assign(`${REACT_APP_OAUTH_PROVIDER_URL}/authorize?${qParams}`);
+   }
+
+   const handleIGLogin = () => {
+      const token =  Cookie.get("igOpenAMRefreshToken") ? Cookie.get("igOpenAMRefreshToken") : null;
+
+      if (token != null) {
+        getOAuthTokens(token)
+          .then(res => {
+             let {access_token} = res;
+             setAccessToken(access_token);
+             setTokenLoaded(true);
+          })
+          .catch(error => {
+             setError(error);
+             console.log(error);
+             redirectToOpenAM();
+          });
+      } else {
+         redirectToOpenAM();
+      }
+   };
+
+   const cleanCookie = () => {
+      Cookie.remove("igOpenAMRefreshToken");
+   };
+
+   useEffect(() => {
+      if (isTokenLoaded) {
+         history.push({
+            pathname: '/oauth',
+            state: { from: location }
+         });
+      }
+   }, [isTokenLoaded]);
+
+   return (
+     <div className="login-container">
+        <Row>
+           <Col></Col>
+           <Col>
+              <div>
+                 <Form>
+                    <FormGroup>
+                       <Button className="login-button" variant="contained" color="primary" onClick={handleIGLogin}>Login with IG</Button>
+                       <Button className="login-button" variant="contained" theme="secondary" style={{marginLeft: "5%" }} onClick={cleanCookie}>Clean IG session</Button>
+                    </FormGroup>
+                 </Form>
+              </div>
```

Comment:
Since we dont actually have any inputs associated with the button, this can just be a button on its own (without Form/FormGroup). To maintain responsiveness, we can oddly use nested `Row` and `Col`.
```html
<Row>
  <Col>
    <Button className="login-button" variant="contained" color="primary" onClick={handleIGLogin}>Login with IG</Button>
  </Col>
  <Col>
    <Button className="login-button" variant="contained" theme="secondary" onClick={cleanCookie}>Clean IG session</Button>
  </Col>
</Row>
```